subsys/net/lib/http/http_server_http1.c

```diff
@@ -256,6 +257,116 @@ static int dynamic_post_req(struct http_resource_detail_dynamic *dynamic_detail,
 	return 0;
 }
 
+#if defined(CONFIG_FILE_SYSTEM)
+
+HTTP_SERVER_CONTENT_TYPE(html, "text/html")
+HTTP_SERVER_CONTENT_TYPE(css, "text/css")
+HTTP_SERVER_CONTENT_TYPE(js, "text/javascript")
+HTTP_SERVER_CONTENT_TYPE(jpg, "image/jpeg")
+HTTP_SERVER_CONTENT_TYPE(png, "image/png")
+HTTP_SERVER_CONTENT_TYPE(svg, "image/svg+xml")
+
+int handle_http1_static_fs_resource(struct http_resource_detail_static_fs *static_fs_detail,
+				    struct http_client_ctx *client)
+{
+#define RESPONSE_TEMPLATE_STATIC_FS                                                                \
+	"HTTP/1.1 200 OK\r\n"                                                                      \
+	"Content-Type: %s%s\r\n\r\n"
+#define CONTENT_ENCODING_GZIP "\r\nContent-Encoding: gzip"
+
+	bool gzipped = false;
+	int len;
+	int remaining;
+	int ret;
+	size_t file_size;
+	struct fs_file_t file;
+	char fname[HTTP_SERVER_MAX_URL_LENGTH];
+	char content_type[HTTP_SERVER_MAX_CONTENT_TYPE_LEN] = "text/html";
+	/* Add couple of bytes to response template size to have space
+	 * for the content type and encoding
+	 */
+	char http_response[sizeof(RESPONSE_TEMPLATE_STATIC_FS) + HTTP_SERVER_MAX_CONTENT_TYPE_LEN +
+			   sizeof(CONTENT_ENCODING_GZIP)];
+	char tmp[TEMP_BUF_LEN];
+
+	if (!(static_fs_detail->common.bitmask_of_supported_http_methods & BIT(HTTP_GET))) {
+		static const char not_found_response[] = "HTTP/1.1 405 Method Not Allowed\r\n"
+							 "Content-Length: 18\r\n\r\n"
+							 "Method Not Allowed";
+
+		ret = http_server_sendall(client, not_found_response,
+					  sizeof(not_found_response) - 1);
+		if (ret < 0) {
+			LOG_DBG("Cannot write to socket (%d)", ret);
+		}
+		return ret;
+	}
+
+	/* get filename and content-type from url */
+	len = strlen(client->url_buffer);
+	if (len > 1) {
```

Comment:
I'm not sure I follow, why special treatment of `len == 1`?

Reply:
`len == 1` means, that the requested path is just the leading `/` and we can just use `index.html` as filename. I added a comment to make that clear.